src/main/java/net/sf/jabref/gui/OpenRemoteDatabaseDialog.java

```diff
@@ -0,0 +1,322 @@
+    private final JPanel buttonPanel = new JPanel();
+
+    private final JLabel databaseTypeLabel = new JLabel(Localization.lang("Database type") + ":");
+    private final JLabel hostPortLabel = new JLabel(Localization.lang("Host") + "/" + Localization.lang("Port") + ":");
+    private final JLabel databaseLabel = new JLabel(Localization.lang("Database") + ":");
+    private final JLabel userLabel = new JLabel(Localization.lang("User") + ":");
+    private final JLabel passwordLabel = new JLabel(Localization.lang("Password") + ":");
+
+    private final JTextField hostField = new JTextField(12);
+    private final JTextField portField = new JTextField(4);
+    private final JTextField userField = new JTextField(14);
+    private final JTextField databaseField = new JTextField(14);
+
+    private final JPasswordField passwordField = new JPasswordField(14);
+    private final JComboBox<DBType> dbTypeDropDown = new JComboBox<>(new DBType[] {DBType.MYSQL, DBType.ORACLE, DBType.POSTGRESQL});
+
+    private final JButton connectButton = new JButton(Localization.lang("Connect"));
+    private final JButton cancelButton = new JButton(Localization.lang("Cancel"));
+
+    private static final String REMOTE_DATABASE_TYPE = "remoteDatabaseType";
+    private static final String REMOTE_HOST = "remoteHost";
+    private static final String REMOTE_PORT = "remotePort";
+    private static final String REMOTE_DATABASE = "remoteDatabase";
+    private static final String REMOTE_USER = "remoteUser";
+
+
+    /**
+     * @param frame the JabRef Frame
+     */
+    public OpenRemoteDatabaseDialog(JabRefFrame frame) {
+        super(frame, Localization.lang("Open remote database"));
+        this.frame = frame;
+        initLayout();
+        applyGlobalPrefs();
+        setupActions();
+        pack();
+    }
+
+    /**
+     * Defines and sets the different actions up.
+     */
+    private void setupActions() {
+
+        Action openAction = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                try {
+                    checkFields();
+                    int port = Integer.parseInt(portField.getText());
+                    BibDatabaseMode selectedMode = Globals.prefs.getDefaultBibDatabaseMode();
+                    DBType selectedType = (DBType) dbTypeDropDown.getSelectedItem();
+
+                    BibDatabaseContext bibDatabaseContext = new BibDatabaseContext(DatabaseLocation.REMOTE,
+                            new Defaults(selectedMode));
+                    DBSynchronizer dbSynchronizer = bibDatabaseContext.getDBSynchronizer();
+
+                    //JPasswordField.getPassword() does not return a String, but a char array.
+                    dbSynchronizer.setUp(DBConnector.getNewConnection(selectedType, hostField.getText(), port,
+                                    databaseField.getText(), userField.getText(), new String(passwordField.getPassword())),
```

Comment:
Please extract the parameters to variables first, e.g. `String password = new String(field.getPassword()`